Задания с 28.09.2021/28.09.2021-Task3/28.09.2021-Task3/Phonebook.c

```diff
@@ -0,0 +1,273 @@
+{
+    Note phoneBook[100] = { 0 };
+    addNote("RazbeR", "+7??????????", phoneBook, 0);
+    addNote("RRB3", "+7??????????", phoneBook, 1);
+    saveData("Test Database.txt", phoneBook, 2);
+    initializeMemory(phoneBook, "Test Database.txt");
+    return strcmp(phoneBook[0].name, "RazbeR") == 0 && strcmp(phoneBook[0].phoneNumber, "+7??????????") == 0 &&
+        strcmp(phoneBook[1].name, "RRB3") == 0 && strcmp(phoneBook[1].phoneNumber, "+7??????????") == 0;
+}
+
+bool addNoteTestPassed(void)
+{
+    Note phoneBook[100] = { 0 };
+    addNote("Vanya Akakievich", "+79876543210", phoneBook, 0);
+    addNote("Petr Gvidonovich", "+71234872092", phoneBook, 1);
+    return strcmp(phoneBook[0].name, "Vanya Akakievich") == 0 && strcmp(phoneBook[0].phoneNumber, "+79876543210") == 0 &&
+        strcmp(phoneBook[1].name, "Petr Gvidonovich") == 0 && strcmp(phoneBook[1].phoneNumber, "+71234872092") == 0;
+}
+
+bool findPhoneNumberTestPassed(void)
+{
+    Note phoneBook[100] = { 0 };
+    const int length = initializeMemory(phoneBook, "Test Database.txt");
+    char phoneNumber[30] = { 0 };
+    findPhoneNumber("RazbeR", phoneNumber, phoneBook, length);
+    return strcmp(phoneNumber, "+7??????????") == 0;
+}
+
+bool findNameTestPassed(void)
+{
+    Note phoneBook[100] = { 0 };
+    const int length = initializeMemory(phoneBook, "Test Database.txt");
+    char name[50] = { 0 };
+    findName("+7??????????", name, phoneBook, length);
+    return strcmp(name, "RazbeR") == 0;
+}
+
+bool generalTestPassed(void)
+{
+    return saveDataTestPassed() && findPhoneNumberTestPassed() && findNameTestPassed() && addNoteTestPassed();
+}
+
+int main(void)
+{
+    clearFile("Test Database.txt");
+    if (!generalTestPassed())
+    {
+        printf("Tests Failed ...\n");
+        return 1;
+    }
+
+    setlocale(LC_ALL, "Russian");
+    Note phoneBook[100] = { 0 };
+    int notesCount = initializeMemory(phoneBook, "Database.txt");
+    int notesCountBeforeSaving = notesCount;
+
+    while (true)
+    {
+        printf("Введите одну из следующих команд:\n");
+        printf("\n");
+        printf("0 - Выйти\n");
+        printf("1 - Добавить новую запись\n");
+        printf("2 - Распечатать все имеющиеся записи\n");
+        printf("3 - Найти телефон по имени\n");
+        printf("4 - Найти имя по телефону\n");
+        printf("5 - Сохранить текущие данные в файл\n");
+
+        int selectedCommand = 0;
+        const int commandInput = scanf("%d", &selectedCommand);
+        while (getchar() != '\n');
+        if (commandInput != 1)
+        {
+            printf("Кажется, вы ввели недопустимые данные, поэтому программа завершает работу ...\n");
+            return 1;
+        }
+
+        if (selectedCommand == 0)
+        {
+            return 0;
+        }
+        else if (selectedCommand == 1)
+        {
+            if (notesCount == 100)
+            {
+                printf("Телефонный справочник переполнен. Больше нельзя добавлять записи.\n");
+            }
```

Comment:
```suggestion
                printf("Телефонный справочник переполнен. Больше нельзя добавлять записи.\n");
                continue;
            }
```
И дальше уже без else